runtime/interpreter/value.go

```diff
@@ -2986,6 +2986,9 @@ func (v *ArrayValue) Reverse(
 	count := v.Count()
 	index := count - 1
 
+	// Meter computation for iterating the array.
+	interpreter.ReportComputation(common.ComputationKindIterateArrayValue, uint(v.Count()))
```

Comment:
Metering up-front works, but might be overly restrictive: If the array is large, the up-front metering cost might already exceed the limit.

It might be good to do the metering like for loops, i.e., meter before each iteration step.

Reply:
It doesn't make much difference really, since all elements of the array are loaded at the time of creation, by the atree.
Anyway updated a07342f so if the atree behavior is changed to not load everything, then it'll reflect here.